src/mesh/smoothing.rs

```diff
@@ -74,12 +74,13 @@ pub fn laplacian_smoothing(
         }
     }
 
-    // FIXME: Calculate smooth normals for the result once we support them
     (
-        Mesh::from_faces_with_vertices_and_normals(
-            mesh.faces().to_vec(),
+        Mesh::from_triangle_faces_with_vertices_and_computed_normals(
```

Comment:
yes, let's do it that way

Reply:
we do a lot of calculation inside `from_triangle_faces_with_vertices_and_computed_normals`. it works because we're not inputting an actual face but merely three numbers. the rest is being calculated.
if we make a generic constructor, we need to make some of decisions, like where should we calculate sharp/smooth normals, what to do with existing normal indices inside the faces?
for now i prefer not doing anything. let's keep it as it is and revisit the issue when we support non-triangular faces.

Reply:
Oh I see the conundrum now :slightly_frowning_face:

Ok. What you did is good enough for now, but we might have to change the structure of our data a bit, because this is ugly. I'll create a trello task for this.